### PR TITLE
Remove the need for ConnectionMonitor in ControlService

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

Instead of using ConnectionMonitor to detect that a connection has been
made. This PR implements 1) A control service which responds to Ping
messages, 2) connection manager sends a ping expecting a pong to
confirm the connection. 3) A control service client which abstracts the
interactions with the control service.

The process for establishing a connection:
1) Connect to peer's control port
2) Send a signed Ping message [new]
3) Wait for a signed Pong [new]
4) If successful, send a signed RequestConnection message 
5) If the node is already trying to connect to the requesting node, a simple deterministic algorithm determines which node's connection attempt should be rejected, while the other one continues [edge_case, new]
6) Await 'Accept' message to peer connection

## Motivation and Context

Ref #445 

## How Has This Been Tested?

Added unit and integration tests for the new message types and control service client 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [x] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by sdbondi** [comment]
WIP: Change to the protocol to prevent possible collisions

File: comms/src/control_service/worker.rs [hunk 1]
```diff
@@ -187,82 +189,278 @@ where
...
-            BindFailed => {
-                self.is_running = false;
-                error!(
+            Err(err) => return Err(ControlServiceError::PeerManagerError(err)),
+        };
+
+        // TODO: SECURITY The node ID is not a verified value at this point (PeerNotFoundError branch above).
+        //       An attacker can insert any node id they want to get information about other peers connections
+        //       to this node. For instance, if they already have an active connection.
+        //       The public key should be used as that is validated by the message signature.
```
Comment (neonknight64) [inline comms/src/control_service/worker.rs]: I agree, we need a mechanism to verify or validate node_ids.

File: comms/src/control_service/worker.rs [hunk 2]
```diff
@@ -187,82 +189,278 @@ where
...
+            ControlServiceMessageType::ConnectRequestOutcome,
+            ConnectRequestOutcome::Accepted,
+        )
+    }
+
+    fn get_peer(&self, public_key: &CommsPublicKey) -> Result<Option<Peer>> {
+        let peer_manager = &self.connection_manager.peer_manager();
+        match peer_manager.find_with_public_key(public_key) {
+            Ok(peer) => Ok(Some(peer)),
+            Err(PeerManagerError::PeerNotFoundError) => Ok(None),
```
Comment (neonknight64) [inline comms/src/control_service/worker.rs]: The PeerManager should probably be changed to rather return a None and not a PeerNotFoundError error.

File: comms/src/control_service/worker.rs (hunk 1 above)
Comment (sdbondi) [reply comms/src/control_service/worker.rs]: Think that validating that the node id is the hash of the public key for base nodes will work - then add a different validation for DAN nodes later - so this issue should be easily resolved

**Pull request merged**